test/testdata/rewriter/minitest_tables.rb

```diff
@@ -86,13 +86,93 @@ def self.it(name, &blk); end
   end
 
 
-  test_each [1, 2, 3] do |k, v| # error: Wrong number of parameters for `test_each` block
-    it "does not handle more than one argument" do
+  test_each [1, 2, 3] do |k, v|
+    it "blocks get padded with NilClass arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+
+    it "multiple it blocks are ok with multiple arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+  end
+
+  test_each [1, 2, 3] do |(k, v)|
+    it "destructured blocks get padded with NilClass arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+
+    it "multiple it blocks are ok with multiple destructured arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+  end
+
+  test_each [[1, ['hi', false]], [2, ['bye', true]]] do |i, (s,b), (x, y)|
+    it "handles mixed destructuring and positional arguments" do
```

Comment:
Should we have at least one test case here with an interpolated `it` string? That is a key user flow in the wild (at least, it's how we found this). Something like:

```ruby
test_each([['red', 2], ['blue', 3]]) do |(color, rank)|
  it "color #{color} which is not purple is not number 1" do
    refute_equal(1, rank)
  end
end
```

Reply:
This exposed a problem with the current implementation: when we make the test name we generate a string based on the AST of the string given to `it`, so your `it` block would have a name like:
```
def <it '::<Magic>.<string-interpolate>("color ", color, " which is not purple is not number 1)'><<todo method>>(&<blk>)
```
The problem with this is that any variables used for string interpolation here that don't support `.to_s` will not have an error raised. I added a test for this and will file a ticket for a follow-on fix.

Reply:
https://github.com/sorbet/sorbet/issues/4524